Src/GoogleApis.Auth.DotNet4/OAuth2/JwtServiceAccountCredential.cs

```diff
@@ -0,0 +1,121 @@
+﻿/*
+Copyright 2013 Google Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Google.Apis.Util;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    /// <summary>
+    /// Service account credential that also supports JSON Web Token access token scenario.
+    /// In this scenario, instead of sending a JWT to a token server and exchanging it for 
+    /// an access token, a slightly different JWT containing the request URI is constructed
+    /// and used as an access token.
+    /// See <see cref="GetAccessTokenForRequestAsync"/> for explanation when JWT access token
+    /// is used and when regular OAuth2 token is used.
+    /// <seealso cref="ServiceAccountCredential"/>
+    /// </para>
+    /// </summary>
+    public class JwtServiceAccountCredential : ServiceAccountCredential
```

Comment:
I also checked with auth guys that ServiceAccountCredential with no scopes is not something useful (because you can't really use Oauth2 token without a scope for anything). So by changing the pre-existing behavior for case with no scopes we will not be breaking anyone. We will just be adding functionality.